cmd/file/io.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package file
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// nopWriteCloser wraps a io.Writer in the same manner as ioutil.NopCloser does
+// with an io.Reader.
+type nopWriteCloser struct {
+	io.Writer
+}
+
+// NopWriteCloser returns a new io.WriteCloser with the given writer as io.Writer.
+func NopWriteCloser(w io.Writer) io.WriteCloser {
+	return &nopWriteCloser{
+		Writer: w,
+	}
+}
+
+// Close implements io.Closer
+func (n *nopWriteCloser) Close() error {
+	return nil
+}
+
+// putGetter wraps both storage.Putter and storage.Getter interfaces
+type putGetter interface {
+	storage.Putter
+	storage.Getter
+}
+
+// TeeStore provides a storage.Putter that can put to multiple underlying storage.Putters.
+type TeeStore struct {
+	putters []storage.Putter
+}
+
+// NewTeeStore creates a new TeeStore.
+func NewTeeStore() *TeeStore {
+	return &TeeStore{}
+}
+
+// Add adds a storage.Putter.
+func (t *TeeStore) Add(putter storage.Putter) {
+	t.putters = append(t.putters, putter)
+}
+
+// Put implements storage.Putter.
+func (t *TeeStore) Put(ctx context.Context, mode storage.ModePut, chs ...swarm.Chunk) (exist []bool, err error) {
+	for _, putter := range t.putters {
+		_, err := putter.Put(ctx, mode, chs...)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return nil, nil
```

Comment:
note you're silently breaking the contract of the function signature. this is not so great as the caller might expect a slice here, resulting in a panic. i would expect at least to return a slice in the same size as the length of the variadic function parameter, all with `false` values